beacon-chain/core/electra/registry_updates.go

```diff
@@ -28,7 +33,37 @@ import (
 //	         for validator in state.validators:
 //	             if is_eligible_for_activation(state, validator):
 //	                 validator.activation_epoch = activation_epoch
-func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state.BeaconState, error) {
-	// TODO: replace with real implementation
-	return state, nil
+func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) error {
+	currentEpoch := time.CurrentEpoch(state)
+	ejectionBal := params.BeaconConfig().EjectionBalance
+	activationEpoch := helpers.ActivationExitEpoch(currentEpoch)
+	vals := state.Validators()
+	for idx, val := range vals {
+		// Handle validators eligible to join the activation queue.
+		if helpers.IsEligibleForActivationQueue(val, currentEpoch) {
+			val.ActivationEligibilityEpoch = currentEpoch + 1
+			if err := state.UpdateValidatorAtIndex(primitives.ValidatorIndex(idx), val); err != nil {
+				return err
+			}
+		}
+		// Handle validator ejections.
+		if val.EffectiveBalance <= ejectionBal && helpers.IsActiveValidator(val, currentEpoch) {
+			var err error
+			// exitQueueEpoch and churn arguments are not used in electra.
+			state, _, err = validators.InitiateValidatorExit(ctx, state, primitives.ValidatorIndex(idx), 0 /*exitQueueEpoch*/, 0 /*churn*/)
```

Comment:
Are these values supposed to be 0?

Reply:
in the function it says it's ignored for electra so I guess it can be anything but 0 seems fine